src/main/java/frc/robot/commands/JsonUpdateCommand.java

```diff
@@ -0,0 +1,59 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.CoordsOutputSubsystem;
+
+public class JsonUpdateCommand extends Command{
+    
+
+    private CoordsOutputSubsystem s_CoordsOutputSubsystem;
+
+    private Timer timer;
+
+    private double timeOffset;
+
+    private int id = 0;
+
+
+    public JsonUpdateCommand() {
+
+        s_CoordsOutputSubsystem = CoordsOutputSubsystem.getInstance();
+
+        timer = new Timer();
+
+    }
+
+    @Override
+    public void initialize() {
+        timeOffset = Timer.getFPGATimestamp();
+    }
+
+    @Override
+    public void execute() {
+        
+        if (timer.hasElapsed(0.5 + timeOffset)) {
+
+            Pose2d botPose = s_CoordsOutputSubsystem.s_Swerve.getPose();
+
+            String coordOutput = "(" + botPose.getX() + ", " + botPose.getY() + ")";
+
+            s_CoordsOutputSubsystem.appendJSON(id, coordOutput);
+            timeOffset = Timer.getFPGATimestamp();
+            id += 0.5;
+
+            System.out.println("JSON OUTPUT: " + id + " : " + botPose);
+
+        } else if (timer.hasElapsed(150)) {
+
+            s_CoordsOutputSubsystem.printJSON();
```

Comment:
As is, this will be called every 20ms after the initial 150 seconds, which will probably cause problems. You could add the hasElapsed check to an isFinished method and then do this in end() instead since this is a Command. However, you may need to change this command to not be the default command for CoordsOutputSubsystem. You could just call schedule() on an instance of it in RobotContainer.

Reply:
ah makes sense

Reply:
done 👍